src/BuildOpt.h

```diff
@@ -1040,14 +1040,37 @@
 
 #if defined(RADIOLIB_DEBUG)
   #if defined(RADIOLIB_BUILD_ARDUINO)
-    #define RADIOLIB_DEBUG_PRINT(...) { RADIOLIB_DEBUG_PORT.print(__VA_ARGS__); }
-    #define RADIOLIB_DEBUG_PRINTLN(...) { RADIOLIB_DEBUG_PORT.println(__VA_ARGS__); }
+// https://github.com/esp8266/Arduino/blob/65579d29081cb8501e4d7f786747bf12e7b37da2/cores/esp8266/Print.cpp#L50
+size_t serialPrintf(const char *format, ...) {
+    va_list arg;
+    va_start(arg, format);
+    char temp[64];
+    char* buffer = temp;
+    size_t len = vsnprintf(temp, sizeof(temp), format, arg);
+    va_end(arg);
+    if (len > sizeof(temp) - 1) {
+        buffer = new char[len + 1];
+        if (!buffer) {
+            return 0;
+        }
+        va_start(arg, format);
+        vsnprintf(buffer, len + 1, format, arg);
+        va_end(arg);
+    }
+    len = RADIOLIB_DEBUG_PORT.write((const uint8_t*) buffer, len);
+    if (buffer != temp) {
+        delete[] buffer;
```

Comment:
Why is buffer being deleted? It's allocated statically.

Reply:
I took this implementation from esp8266 Arduino (cf. the link)

Reply:
I didn't notice the dynamic allocation a couple lines above, in that case it's OK.